src/types.ts

```diff
@@ -0,0 +1,161 @@
+import Key from './key'
+
+export type AnyIterable<T> = Iterable<T> | AsyncIterable<T>
+export type PromiseOrValue<T> = Promise<T> | T
```

Comment:
Likewise I called this `Await<T>` in other places, again implying that you need to `await` it to get a `T`.